pandas/io/excel.py

```diff
@@ -66,27 +69,211 @@ def get_writer(engine_name):
+        file reader class that inherits from BaseFile, withouth failing, in
+        order to be able to deduce which file type based on a path extension.
+        """
+        self.engine = engine
+        self.io_class = None
+        self._open_workbook = None
+        # fail silently if engine is not presented on init, only raise when
+        # attempting to use it when opening a workbook
+        try:
+            self.load_engine()
+        except ImportError:
+            pass
+#        if io is not None:
+#            self.engine, self.book = self.create(io, engine=engine)
+#        return self.engine, self.book
+
+    def open_workbook(self, *args, **kwargs):
+        """Explicitely load the engine again (and trigger an ImportError in the
+        process) if _open_workbook is set to None.
+        """
+        # try to load the engine again and raise import error if required
+        if self._open_workbook is None:
+            self.load_engine()
+        # just in case the user passes an already opened workbook of io_class
+        if len(args) > 0 and isinstance(args[0], self.io_class):
+            return args[0]
+        return self._open_workbook(*args, **kwargs)
+
+    def is_ext(self, path):
+        """Verify if the path's extension is supported by the reader. If the
+        reader is not installed, return False
+        """
+        ext = path.split('.')[-1]
+        if self.io_class is None:
+            return False
+        return ext in self.extensions
+
+    def load_engine(self):
+        """Load the engine if installed, but raise ImportError otherwise.
+        If engine is not defined, do nothing.
+        """
+        try:
+            _readers[self.engine] = self._load_engine()
+        except AttributeError:
+            _readers[self.engine] = self._load_engine()
+            msg = 'Excel reader engine "%s" is not implemented' % self.engine
+            raise NotImplementedError(msg)
+
+    def _load_engine(self):
+        return False
+
+    def create(self, io, engine=None):
+        """Create the appropriate speadsheet book based on io and optionally
+        engine.
+
+        Paratemeters
+        ------------
+        io : string, file-like object or xlrd/ezodf workbook
+            If a string, expected to be a path to xls, xlsx, or ods file.
+            File-like objects or buffers are only supported for xlrd types.
+
+        engine: string, default None
+            If io is not a buffer or path, this must be set to identify io.
+            Acceptable values are None, xlrd, or ezodf
+
+        Returns
+        -------
+        engine : string
+            Engine used for reading the io
+
+        book : object
+            The spreadsheet book class created by the engine.
+        """
+        self.engine = engine
+        # Initiate pandas supported file readers. When creating new readers,
+        # add them to the supported_readers list
+        xlrd_f = XLRDFile()
+        ezodf_f = EZODFFile()
```

Comment:
why don't you make ``supported_readers`` a cached property which returns this list. This should return readers which don't raise import errors.